Vultr API v2

excid3 opened the pull request:
This is a complete rewrite of the gem with major improvements:

1. API v2 support
2. Responses are converted to `Vultr` objects using OpenStruct (for example `Vultr::Instance`)
3. Tests for each endpoint

We just have a few more tests left to add before this is ready to merge and release. excid3/vultr.rb#3

excid3 committed:
WIP

excid3 committed:
Add more resources

afomera committed:
Update more of the API endpoints and that should bring things to be mostly functional

Co-authored-by: Chris Oliver <excid3@gmail.com>

afomera committed:
Install and run Standardrb

Co-authored-by: Chris Oliver <excid3@gmail.com>

afomera committed:
Add CI workflow and bump version num to 1.0

Co-authored-by: Chris Oliver <excid3@gmail.com>

afomera committed:
Lint and define respond_to_missing?

Co-authored-by: Chris Oliver <excid3@gmail.com>

excid3 committed:
Add all endpoints to docs

excid3 committed:
Rename backup schedule

excid3 committed:
Add bin scripts

excid3 committed:
Okay, we'll use the keyword args @afomera. Fine.

excid3 committed:
Version bump

BilalBudhani committed:
Updates Vultr::Object to support marshalling

BilalBudhani committed:
Adds test case for number

excid3 committed:
Merge pull request #2 from BilalBudhani/vultr-object

Marshalling Support for Vultr Object

excid3 committed:
Add test stubs

excid3 committed:
Merge pull request #1 from excid3/v2-api

Vultr API Version 2

excid3 committed:
Update readme

excid3 committed:
Update badge and install instructions to point to fork

excid3 committed:
Merge branch 'v2-api'

leoanthonyc committed:
Add tests for startup_scripts resource

excid3 committed:
Merge pull request #10 from leoanthonyc/add-startup-scripts-resource-test

Add tests for startup_scripts resource

stevepolitodesign committed:
Add fixture.

https://www.vultr.com/api/#operation/list-applications

stevepolitodesign committed:
Write tests for Applications Resource.

Simple test for the list endpoint inspired by the existing test created
for the Users Resources list endpoint.

stevepolitodesign committed:
Add fixture data.

https://www.vultr.com/api/#operation/list-backups
https://www.vultr.com/api/#operation/get-backup

stevepolitodesign committed:
Write tests for Backups Resource

Simple tests for the list and retrieve endpoints inspired by the existing test created
for the Users Resources list and retrieve endpoints.

Commit:
Fix: remove string variable

Commit:
add plans tests

- [x] list
- [x] list_metal

excid3 committed:
Merge pull request #15 from JuanVqz/add_plans_tests

add plans tests

excid3 committed:
Update version.rb

excid3 committed:
Merge pull request #13 from stevepolitodesign/backups-resource

Test Backups Resource

excid3 committed:
Merge pull request #12 from stevepolitodesign/applications-resource

Test Applications Resource

Commit:
add DNS tests (#16)

* add dns test

references #3

* Update dns_test.rb

Co-authored-by: Chris Oliver <excid3@gmail.com>

Commit:
Add tests for ssh_keys resource (#11)

excid3 committed:
Merge branch 'remove_string_variable' of https://github.com/JuanVqz/vultr.rb into JuanVqz-remove_string_variable

excid3 committed:
Update gemspec to match

excid3 committed:
Merge branch 'JuanVqz-remove_string_variable'

Commit:
Add tests for operating_systems section. (#17)

Commit:
Add tests for Regions resource. (#18)

Commit:
Add tests for snapshots resource (#19)

Commit:
Add tests for reserved_ips resource (#20)

tolbkni commented:
Good job! To @excid3, you should have the commit permission of the repo now, so maybe you can merge this by yourself.

Commit:
Add tests for object storage and private networks (#21)

* Add tests for private_networks resource

* Add tests for object_storage resource

excid3 committed:
Recursively convert to OpenStruct instead of converting to and from JSON redundantly

excid3 commented:
Thanks a bunch @tolbkni and thanks for making this gem! 🙏

tolbkni commented on vultr.gemspec:
```diff
@@ -1,26 +1,26 @@
+  s.require_paths = ["lib"]
 
-  s.required_ruby_version = '>= 1.9'
+  s.required_ruby_version = ">= 1.9"
```
@excid3 Maybe we can bump the ruby version from ">= 1.9" to ">= 2.6", because the versions under 2.6 are EOL.

excid3 replied:
Okay! Just checked and Faraday 1.7 requires Ruby >= 2.4, so we'll use that as our minimum version.

Faraday 2.0 is just around the corner and requires Ruby >= 2.6. I don't see a need to use that yet.

tolbkni commented on lib/vultr/version.rb:
```diff
@@ -1,10 +1,3 @@
-
-    STRING = [MAJOR, MINOR, TINY, PRE].compact.join('.')
-  end
+  VERSION = "2.0.0"
```
@excid3 Why we skip the 1.0 version and bump to 2.0 directly?

excid3 replied:
I thought since we're implement API v2, maybe we just have the major version match?

excid3 replied:
Changed this to 1.0. I don't think we need to track the same version as their API. 👍

excid3 committed:
Try testing old rubies

Commit:
add ISO test (#22)

reference #3

excid3 committed:
Update ruby versions for CI

excid3 committed:
Merge branch 'master' of github.com:excid3/vultr.rb

excid3 committed:
Update minimum Ruby dependency to match Faraday

excid3 committed:
Change to version 1.0.0

excid3 committed:
Add rate limit status handling

partydrone committed:
Use Faraday middleware to include Authorization header (#23)

* Use OAuth2 middleware to include Authorization header

* Add Authorization header to client configuration

Adding directly instead of using the OAuth2 middleware since this gem technically doesn't use OAuth.

* Remove default_headers method

It's no longer needed now that we set the Authorization header in Faraday.

Co-authored-by: Chris Oliver <excid3@gmail.com>

excid3 committed:
Merge pull request #24 from excid3/v2-api

Use Faraday middleware to include Authorization header (#23)

excid3 committed:
Use recursive open struct for Objects

excid3 committed:
Dnssec should return array of strings, not an object

excid3 committed:
standardize

partydrone committed:
Parse ISO8601 date strings into `Time` objects

feipinghuang committed:
Update block_storage.rb

excid3 committed:
Merge pull request #25 from partydrone/parse-dates

Parse ISO8601 date strings into `Time` objects

excid3 committed:
Merge pull request #26 from feipinghuang/patch-1

Update block_storage.rb

ashwin47 committed:
Tests for Instances Resource (#27)

* added tests for instances resourse

* standardb

* Fix instance neighbors

* Update instances_test.rb

* standardb fixes

Co-authored-by: Chris Oliver <excid3@gmail.com>

Commit:
Add tests for load_balancers resource (#28)

excid3 committed:
Add firewall tests

gordysc committed:
Add tests for bare_metals resource (#30)

* Add bare metal tests (#1)

* test(bare_metal): test list

* test(bare_metal): test create

* fix(bare_metal): correcting typo from doc

* test(bare_metal): test #retrieve

* test(bare_metals): #update

* test(bare_metals): delete

* test(bare_metals): start

* test(bare_metals): reboot

* test(bare_metals): reinstall

* test(bare_metals): halt

* test(bare_metals): bandwidth

* test(bare_metal): cleaning up assertions

* test(bare_metals): user_data

* test(bare_metals): upgrades

* test(bare_metals): vnc

* test(bare_metals): list_ipv4

* test(bare_metals): list_ipv6

* test(bare_metals): halt_instances

* test(bare_metals): reboot_instances

* test(bare_metals): start_instances

* fix(bare_metal): addressing styling errors

gordysc committed:
Resource tests for Kubernetes & Block Storage (#32)

* Add bare metal tests (#1)

* test(bare_metal): test list

* test(bare_metal): test create

* fix(bare_metal): correcting typo from doc

* test(bare_metal): test #retrieve

* test(bare_metals): #update

* test(bare_metals): delete

* test(bare_metals): start

* test(bare_metals): reboot

* test(bare_metals): reinstall

* test(bare_metals): halt

* test(bare_metals): bandwidth

* test(bare_metal): cleaning up assertions

* test(bare_metals): user_data

* test(bare_metals): upgrades

* test(bare_metals): vnc

* test(bare_metals): list_ipv4

* test(bare_metals): list_ipv6

* test(bare_metals): halt_instances

* test(bare_metals): reboot_instances

* test(bare_metals): start_instances

* fix(bare_metal): addressing styling errors

* Kubernetes resource tests (#2)

* test(kubernetes): assertions for #list

* test(kubernetes): assertions for #create

* test(kubernetes): assertions for #retrieve

* test(kubernetes): assertions for #update

* test(kubernetes): assertions for #delete

* test(kubernetes): assertions for #config

* test(kubernetes): assertions for #list_resources

* test(kubernetes): assertions for #list_node_pools

* test(kubernetes): assertions for #retrieve_node_pool

* test(kubernetes): assertions for #create_node_pool

* test(kubernetes): assertions for #update_node_pool

* test(kubernetes): assertions for #delete_node_pool

* test(kubernetes): assertions for #delete_node_pool_instance

* test(kubernetes): assertions for #recycle_node_pool_instance

* fix(kubernetes): address style errors

* fix(kubernetes): adding pagination to #list_node_pools

* test(block_storage): covers all actions for resource

* fix(block_storage): correcting style errors

* fix(blocks): remove patch file

JuanVqz commented:
great work! thank you for allowing us to contribute.
